elastic/kibana
# [Reporting Management] Consolidate report job warnings and add warning for deprecated types

## Summary

Closes https://github.com/elastic/kibana/issues/106154
Closes https://github.com/elastic/kibana/issues/100363

## Release note:
There are changes to the Reporting page under Stack Management. Any report job that has completed with warnings will now show all the warnings in a popover, same as the failure message for jobs that have failed. There is a new type of warning for "deprecated" export types. Deprecated export types of reports can only show in the list of jobs if you have an automated report that was created in an older version of Kibana, and the warning will tell you to redo the automation with a new POST URL in order for the report to keep working in future versions.

## List of changes:
 - Takes the "Max size reached" message out of the report listing table. 
 - Removes specific warnings in the "hover" tooltips
 - Adds `EuiPopover` elements that show all warnings (or the failure message) in one place
 - Adds reports with varying statuses to the `reporting/archived_reports` ES test archive
 - Changed the "error button" to show the error message without a call to an API
 - Removes code for a non-existent `cancelled` status
 - Includes various improvements for type strictness
 - Minor renaming of props/variable names ("record" -> "job") to match the TS type
 - Changes language of "Job" to "Report" in the user-facing text.
 - Fixed some user facing text that was not wrapped in a translation function

Removed API endpoint:
 - Removed the `/api/reporting/output/<jobId>` (private API) which was inefficiently being used to fetch the error message of a job. The same can be achieved using `/api/reporting/info/<jobId>` since the returned data type includes warnings, which for a failed job, is the failure message.

## How to test with the updated archived reports
You can start Kibana in normal dev mode, and load the archive of various kinds of report jobs, which is being used in functional testing:
```sh
> node scripts/es_archiver.js \
  --config x-pack/test/functional/config.js \
  load x-pack/test/functional/es_archives/reporting/archived_reports \
  --es-url https://elastic:changeme@localhost:9200 \
  --es-ca config/cert-bundle/ca/ca.crt \ #  (use this example if you run Elasticsearch over HTTPS)
  --kibana-url http://elastic:changeme@localhost:5601
```

Sign in as `test_user` to see the archived reports show up in Stack Management > Reporting.

### Screenshots
**List of reports from the `reporting/archived_reports` archive:
Before/After**
![image](https://user-images.githubusercontent.com/908371/126725730-01286c4f-7c62-4f58-b357-8aa421d77776.png)
![image](https://user-images.githubusercontent.com/908371/126725984-dc43c6f8-836b-46c1-a54d-591881769479.png)

**List of reports from the `reporting/archived_reports` archive showing pending and processing reports:
Before/After**
![image](https://user-images.githubusercontent.com/908371/126725540-845e5784-dda5-4cd7-8f8c-ec67b1cd81ce.png)
![image](https://user-images.githubusercontent.com/908371/126725440-68f85c41-75ed-4d2d-9e4f-60071cfa1bb4.png)


**Note:** The UI changes to the table are to give more focus to the errors and warnings in a consistent way. There is a separate issue for redesigning the table to manage reports, which is still unstarted: https://github.com/elastic/kibana/issues/39620


### Checklist

Delete any items that are not applicable to this PR.
